internal/regexpchecker/regexpchecker.go

```diff
@@ -0,0 +1,1351 @@
+			}
+			suggestion := core.GetSpellingSuggestion(propertyNameOrValue, candidates, core.Identity[string])
+			if suggestion != "" {
+				v.error(diagnostics.Did_you_mean_0, propertyNameOrValueStart, v.pos-propertyNameOrValueStart, suggestion)
+			}
+			propertyNameValid = false
+		}
+		v.pos++
+		propertyValueStart := v.pos
+		v.scanIdentifier(v.charAtOffset(0))
+		propertyValue := v.tokenValue
+		if v.pos == propertyValueStart {
+			v.error(diagnostics.Expected_a_Unicode_property_value, propertyValueStart, 0)
+		} else if propertyNameValid && !isValidUnicodeProperty(propertyNameOrValue, propertyValue) {
+			v.error(diagnostics.Unknown_Unicode_property_value, propertyValueStart, v.pos-propertyValueStart)
+			// Provide spelling suggestion based on the property name
+			canonicalName := nonBinaryUnicodePropertyNames[propertyNameOrValue]
+			var candidates []string
+			if canonicalName == "General_Category" {
+				candidates = generalCategoryValues.KeysSlice()
+			} else if canonicalName == "Script" || canonicalName == "Script_Extensions" {
+				candidates = scriptValues.KeysSlice()
+			}
+			if len(candidates) > 0 {
+				suggestion := core.GetSpellingSuggestion(propertyValue, candidates, core.Identity[string])
+				if suggestion != "" {
+					v.error(diagnostics.Did_you_mean_0, propertyValueStart, v.pos-propertyValueStart, suggestion)
+				}
+			}
+		}
+	} else {
+		// property name alone
+		if v.pos == propertyNameOrValueStart {
+			v.error(diagnostics.Expected_a_Unicode_property_name_or_value, propertyNameOrValueStart, 0)
+		} else if binaryUnicodePropertiesOfStrings.Has(propertyNameOrValue) {
+			// Properties that match more than one character (strings)
+			if !v.unicodeSetsMode {
+				v.error(diagnostics.Any_Unicode_property_that_would_possibly_match_more_than_a_single_character_is_only_available_when_the_Unicode_Sets_v_flag_is_set, propertyNameOrValueStart, v.pos-propertyNameOrValueStart)
+			} else if isCharacterComplement {
+				v.error(diagnostics.Anything_that_would_possibly_match_more_than_a_single_character_is_invalid_inside_a_negated_character_class, propertyNameOrValueStart, v.pos-propertyNameOrValueStart)
+			} else {
+				v.mayContainStrings = true
+			}
+		} else if !isValidUnicodePropertyName(propertyNameOrValue) {
+			v.error(diagnostics.Unknown_Unicode_property_name_or_value, propertyNameOrValueStart, v.pos-propertyNameOrValueStart)
+			// Provide spelling suggestion from general category values, binary properties, and binary properties of strings
+			candidates := make([]string, 0, generalCategoryValues.Len()+binaryUnicodeProperties.Len()+binaryUnicodePropertiesOfStrings.Len())
+			candidates = slices.AppendSeq(candidates, maps.Keys(generalCategoryValues.M))
+			candidates = slices.AppendSeq(candidates, maps.Keys(binaryUnicodeProperties.M))
+			candidates = slices.AppendSeq(candidates, maps.Keys(binaryUnicodePropertiesOfStrings.M))
+			suggestion := core.GetSpellingSuggestion(propertyNameOrValue, candidates, core.Identity)
```

Comment:
Did you consider making `GetSpellingSuggestion` accept an iterator to avoid allocation? (I am not particularly familiar with Go; is the succession of hidden `for` loops unavoidable after all?)